### DISCUSSION TITLE
HDR 转 SDR 的优化方案

### DISCUSSION THREAD

**Discussion by natural-harmonia-gropius** [hooke007/mpv_PlayKit#272] [INFO 信息]
**本文以 `vo=gpu-next` 为必要条件**

目前 mpv 的默认参数我认为没什么可调整的
如果觉得亮度总是在跳动那就在 mpv.conf 中增加 `hdr-peak-percentile=99.995` 来忽略一部分峰值（更低的值忽略的越多，个人建议不低于 99.7），或使用 `hdr-compute-peak=no` 直接关闭峰值检测
如果觉得高亮部分细节不足可以尝试 `hdr-contrast-recovery=0.30` 和 `hdr-contrast-smoothness=3.5`

## Shader

但是我并不满意 mpv 的映射结果（饱和度略高），所以自己做了转换着色器
https://github.com/natural-harmonia-gropius/hdr-toys

缺点是界面和字幕会从 sRGB 转成与视频相同的高动态范围、广色域色彩空间，导致看起来非常暗淡

**Comment by natural-harmonia-gropius** [comment]


**Reply by dyphire** [reply]
--vo=gpu-next 新实现的 st2094 映射算法的个人看法参考：https://github.com/hooke007/MPV_lazy/discussions/39#discussioncomment-4966121

**Reply by dyphire** [reply]
感觉没有什么必要性了。~~gpu-next下的hdr参数除了色域映射目前存在问题需要将`gamut-mapping-mode`切换为clip外，~~ 其他选项在启用`hdr-compute-peak`峰值检测后保持默参下的动态映射效果已经足够好了，没什么值得调整的（profile=gpu-hq的几个hdr参数感兴趣可以尝试调整

对部分影片的hdr动态映射效果不满意的话，想改善可能也只有上你的着色器转换方案了

**Comment by hooke007** [comment]
# mpv-lazy-2025V1 快速适配流程

使用的 hdr-toys 版本 https://github.com/natural-harmonia-gropius/hdr-toys/releases/tag/v2502

## 1. 修改预设的冲突参数

移除选项保存项目中的着色器条目，编辑 script-opts.conf
```ini
 script-opts-append = save_global_props-props=volume,mute
```

移除原tonemap设计，编辑 profiles.conf
```ini
# 删除这部分的整个区块
[HDR_generic]
 profile-desc ......
 ......
 ......
```

## 2. 安装 hdr-toys

下载上文指定的 hdr-toys 版本，将各个文件放置到对应目录中。

新增追加读取一条配置文件，编辑 mpv.conf
```ini
 include = "~~/profiles.conf"
 include = "~~/script-opts.conf"
 include = "~~/hdr-toys.conf"
```

播放 hdr 视频，即可正常激活。

**Reply by hooke007** [reply]
不要在此层回复。
如遇到适配问题请在版本专帖下回复。
